### PR TITLE
Ideep converge

### PR DESCRIPTION
**Pull request by Guobing-Chen**
This PR converge the files of abstract_types.hpp/eltwise.hpp/utils.hpp/tensor.hpp.

Details of converge notes are as below:
https://intel.sharepoint.com/:w:/s/ProductPerformance/EUCTYHXaNblFgjCNorpBYioBB0RjOizwkhWQ0pLT16Pcgg?e=JrEi42

### CODE REVIEW

**Comment by Guobing-Chen** [comment]
@jgong5 , @EikanWang , do you have any review comments on this PR?

File: include/ideep/tensor.hpp [hunk 1]
```diff
@@ -259,6 +263,30 @@ class tensor : public memory {
           strides[w] == 1;
     };
 
+    inline bool is_channels_last() const {
```
Comment (jgong5) [inline include/ideep/tensor.hpp:266]: Do you assume `caffe2_use` is true in this implementation?
https://github.com/Guobing-Chen/ideep/blob/pytorch_dnnl/include/ideep/tensor.hpp#L221

Comment (Guobing-Chen) [reply include/ideep/tensor.hpp:266]: Regarding is_channels_last(), as from the code history, it comes from IPEX/Ideep and only used in IPEX/Ideep Conv OP. Caffe2 does not use is_channels_last(), while does use is_nhwc()/is_nchw().

Comment (Guobing-Chen) [reply include/ideep/tensor.hpp:266]: @jgong5 , any more suggestion for this PR?

**Review by jgong5 — APPROVED** [review]

**Pull request merged**